### PR TITLE
[Api] [Checkout] skipping shipment step

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | based on #11705 

| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Resources/config/suites/api/checkout/checkout.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ default:
                 - sylius.behat.context.transform.lexical
                 - sylius.behat.context.transform.order
                 - sylius.behat.context.transform.payment
+                - sylius.behat.context.transform.payment
```
Comment (arti0090) [inline src/Sylius/Behat/Resources/config/suites/api/checkout/checkout.yml:16]: Looks like a duplicate entry 🤔

File: features/checkout/skipping_shipping_step_when_only_one_shipping_method_is_available.feature [hunk 2]
```diff
@@ -23,7 +23,7 @@ Feature: Skipping shipping step when only one shipping method is available
         Then I should be on the checkout complete step
         And my order's shipping method should be "DHL"
 
-    @ui
+    @api
```
Comment (GSadee) [inline features/checkout/skipping_shipping_step_when_only_one_shipping_method_is_available.feature:26]: `@ui` shouldn't be removed

File: src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutCompleteContext.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    public function iProvideAdditionalNotesLike(string $notes): void
+    {
+        $this->content['additionalNote'] = $notes;
+
+        $this->sharedStorage->set('additional_note', $notes);
+    }
+
+    /**
+     * @Then I should be on the checkout complete step
+     * @Then I should be on the checkout summary step
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutCompleteContext.php:61]: `Then` steps should be after `When`

File: src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutShippingContext.php [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    ) {
+        $this->client = $client;
+        $this->responseChecker = $responseChecker;
+        $this->sharedStorage = $sharedStorage;
+        $this->shippingMethodRepository = $shippingMethodRepository;
+    }
+
+    /**
+     * @When I complete the shipping step
+     * @When And I complete the shipping step
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutShippingContext.php:54]:
```suggestion
```

File: src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutShippingContext.php [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        return $headers;
+    }
+
+    private function getCheckoutState(): string
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+
+        return $this->responseChecker->getValue($response, 'checkoutState');
+    }
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/Checkout/CheckoutShippingContext.php:131]: These private methods are implemented in every `Checkout*Context.php`

**Comment by GSadee** [comment]
Thank you, Adam! :1st_place_medal:

**Pull request merged**